core/src/components/segment/segment.tsx

```diff
@@ -78,11 +89,261 @@ export class Segment implements ComponentInterface {
     this.emitStyle();
   }
 
-  componentDidLoad() {
+  componentWillLoad() {
+    this.emitStyle();
+  }
+
+  async componentDidLoad() {
     this.updateButtons();
+    this.setCheckedClasses();
+
+    this.gesture = (await import('../../utils/gesture')).createGesture({
+      el: this.el,
+      gestureName: 'segment',
+      gesturePriority: 100,
+      threshold: 0,
+      passive: false,
+      onStart: ev => this.onStart(ev),
+      onMove: ev => this.onMove(ev),
+      onEnd: ev => this.onEnd(ev),
+    });
+    this.disabledChanged();
+
     this.didInit = true;
   }
 
+  onStart(detail: GestureDetail) {
+    this.activate(detail);
+  }
+
+  onMove(detail: GestureDetail) {
+    this.setNextIndex(detail);
+  }
+
+  onEnd(detail: GestureDetail) {
+    this.activated = false;
+
+    this.setNextIndex(detail, true);
+
+    detail.event.preventDefault();
+    detail.event.stopImmediatePropagation();
+
+    const ripple = (detail.event!.target! as HTMLElement).shadowRoot!.querySelector('ion-ripple-effect');
+    if (!ripple) { return; }
+
+    const { x, y } = pointerCoord(detail.event);
+
+    ripple.addRipple(x, y).then(remove => remove());
+  }
+
+  private activate(detail: GestureDetail) {
+    const clicked = detail.event.target as HTMLIonSegmentButtonElement;
+    const buttons = this.getButtons();
+    const checked = buttons.find(button => button.checked === true);
+
+    // Make sure we are only checking for activation on a segment button
+    // since disabled buttons will get the click on the segment
+    if (clicked.tagName !== 'ION-SEGMENT-BUTTON') {
+      return;
+    }
+
+    // If there are no checked buttons, set the current button to checked
+    if (!checked) {
+      clicked.checked = true;
+    }
+
+    // If the gesture began on the clicked button with the indicator
+    // then we should activate the indicator
+    if (clicked.checked) {
+      this.activated = true;
+    }
+
+    this.activateIndicator();
+  }
+
+  private activateIndicator() {
+    const activated = this.activated;
+    const buttons = this.getButtons();
+    const clicked = buttons.find(button => button.checked === true);
+    const mode = getIonMode();
+
+    if (!clicked) {
+      return;
+    }
+
+    const indicator = this.getIndicator(clicked);
+
+    if (!indicator) {
+      return;
+    }
+
+    const reduceMotion = window.matchMedia('(prefers-reduced-motion: reduce)').matches;
+
+    if (activated && mode === 'ios' && !reduceMotion) {
+      indicator.style.setProperty('transform', 'scale(0.95)');
+    } else {
```

Comment:
Are we able to do this from the CSS? Maybe use a css var or something:

`indicator.style.setProperty('transform', 'var(--indicator-scale)');`

Reply:
Doing this now using CSS only